--- htdocs.orig/Bugzilla/BugMail.pm	2005-05-25 13:05:01.000000000 +0300
+++ htdocs/Bugzilla/BugMail.pm	2006-01-19 13:22:36.000000000 +0200
@@ -54,6 +54,8 @@
 my %seen;
 my @sentlist;
 
+
+
 # I got sick of adding &:: to everything.
 # However, 'Yuck!'
 # I can't require, cause that pulls it in only once, so it won't then be
@@ -400,6 +402,16 @@
 
         }
     }
+    # baris: have no idea how this works.
+    my @r;
+    push(@r, 'CC');
+    NewProcessOnePersonForBugzilla("bugzilla\@uludag.org.tr", 0, \@headerlist,
+                                         \@r, \%values,
+                                         \%defmailhead, 
+                                         \%fielddescription, \@diffparts,
+                                         $newcomments, 
+                                         $anyprivate, $start, $id, 
+                                         \@depbugs);
 
 
     SendSQL("UPDATE bugs SET lastdiffed = '$end', delta_ts = delta_ts " .
@@ -674,6 +686,125 @@
     return @recipients;
 }
 
+
+# Baris: Dirty. But perl is dirty...
+sub NewProcessOnePersonForBugzilla ($$$$$$$$$$$$$) {
+    my ($person, $count, $hlRef, $reasonsRef, $valueRef, $dmhRef, $fdRef,  
+        $diffRef, $newcomments, $anyprivate, $start, 
+        $id, $depbugsRef) = @_;
+
+    my %values = %$valueRef;
+    my @headerlist = @$hlRef;
+    my @reasons = @$reasonsRef;
+    my %defmailhead = %$dmhRef;
+    my %fielddescription = %$fdRef;
+    my @diffparts = @$diffRef;
+    my @depbugs = @$depbugsRef;
+    
+    my %mailhead = %defmailhead;
+    
+    my $head = "";
+    
+    foreach my $f (@headerlist) {
+      if ($mailhead{$f}) {
+        my $value = $values{$f};
+        # If there isn't anything to show, don't include this header
+        if (! $value) {
+          next;
+        }
+        # Only send estimated_time if it is enabled and the user is in the group
+        if ($f ne 'estimated_time') {
+            my $desc = $fielddescription{$f};
+            $head .= FormatDouble($desc, $value);
+        }
+      }
+    }
+
+    # Build difftext (the actions) by verifying the user should see them
+    my $difftext = "";
+    my $diffheader = "";
+    my $add_diff;
+    $add_diff = 0;
+        
+
+    if ($difftext eq "" && $newcomments eq "") {
+      # Whoops, no differences!
+      return;
+    }
+    
+    my $reasonsbody = "------- You are receiving this mail because: -------\n";
+
+    if (scalar(@reasons) == 0) {
+        $reasonsbody .= "Whoops!  I have no idea!\n";
+    } else {
+        foreach my $reason (@reasons) {
+            if ($reason eq 'AssignedTo') {
+                $reasonsbody .= "You are the assignee for the bug, or are watching the assignee.\n";
+            } elsif ($reason eq 'Reporter') {
+                $reasonsbody .= "You reported the bug, or are watching the reporter.\n";
+            } elsif ($reason eq 'QAcontact') {
+                $reasonsbody .= "You are the QA contact for the bug, or are watching the QA contact.\n";
+            } elsif ($reason eq 'CC') {
+                $reasonsbody .= "You are on the CC list for the bug, or are watching someone who is.\n";
+            } elsif ($reason eq 'Voter') {
+                $reasonsbody .= "You are a voter for the bug, or are watching someone who is.\n";
+            } else {
+                $reasonsbody .= "Whoops!  There is an unknown reason!\n";
+            }
+        }
+    }
+
+    my $isnew = ($start !~ m/[1-9]/);
+    
+    my %substs;
+
+    # If an attachment was created, then add an URL. (Note: the 'g'lobal
+    # replace should work with comments with multiple attachments.)
+
+    if ( $newcomments =~ /Created an attachment \(/ ) {
+
+        my $showattachurlbase =
+            Param('urlbase') . "attachment.cgi?id=";
+
+        $newcomments =~ s/(Created an attachment \(id=([0-9]+)\))/$1\n --> \(${showattachurlbase}$2&action=view\)/g;
+    }
+
+    $person .= Param('emailsuffix');
+# 09/13/2000 cyeh@bluemartini.com
+# If a bug is changed, don't put the word "Changed" in the subject mail
+# since if the bug didn't change, you wouldn't be getting mail
+# in the first place! see http://bugzilla.mozilla.org/show_bug.cgi?id=29820 
+# for details.
+    $substs{"neworchanged"} = $isnew ? ' New: ' : '';
+    $substs{"to"} = $person;
+    $substs{"cc"} = '';
+    $substs{"bugid"} = $id;
+    if ($isnew) {
+      $substs{"diffs"} = $head . "\n\n" . $newcomments;
+    } else {
+      $substs{"diffs"} = $difftext . "\n\n" . $newcomments;
+    }
+    $substs{"summary"} = $values{'short_desc'};
+    use Encode;
+    $substs{"summary"} = encode('MIME-Q', $substs{"summary"});
+    $substs{"reasonsheader"} = "";
+    $substs{"reasonsbody"} = "";
+    $substs{"space"} = " ";
+    
+    my $template = Param("newchangedmail");
+    
+    my $msg = PerformSubsts($template, \%substs);
+
+    MessageToMTA($msg, $person);
+
+    push(@sentlist, $person);
+    return 1;
+}
+
+
+
+
+
 sub NewProcessOnePerson ($$$$$$$$$$$$$) {
     my ($person, $count, $hlRef, $reasonsRef, $valueRef, $dmhRef, $fdRef,  
         $diffRef, $newcomments, $anyprivate, $start, 
