From 462620823a06791bebceccfff51369b868d6cd68 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Caio=20Rom=C3=A3o?= <caioromao@gmail.com>
Date: Sun, 14 Jun 2009 20:43:35 -0300
Subject: [PATCH 03/10] [FIX] IndexError when searching for non-existant keyword.

---
 bugz/bugzilla.py |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/bugz/bugzilla.py b/bugz/bugzilla.py
index d2b1e97..8e5fdc6 100644
--- a/bugz/bugzilla.py
+++ b/bugz/bugzilla.py
@@ -293,6 +293,9 @@ class Bugz:
 				self.log('Unknown field: ' + field)
 				columns.append(field)
 		for row in rows[1:]:
+			if len(row) == 0:
+				self.log('Bugzilla error (row is empty)')
+				return None
 			if row[0].find("Missing Search") != -1:
 				self.log('Bugzilla error (Missing search found)')
 				return None
-- 
1.6.3.2

