From 5763c3ad15c20cb308d57092ec65d924542f626d Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Caio=20Rom=C3=A3o?= <caioromao@gmail.com>
Date: Wed, 17 Jun 2009 17:51:56 -0300
Subject: [PATCH 09/10] cleanup

---
 bugz/bugzilla.py |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/bugz/bugzilla.py b/bugz/bugzilla.py
index 99ac613..edf9bc0 100644
--- a/bugz/bugzilla.py
+++ b/bugz/bugzilla.py
@@ -648,9 +648,6 @@ class Bugz:
 
         try:
             resp = self.opener.open(req)
-            if duplicate:
-                print qparams
-                print req_params
             return modified
         except:
             return []
-- 
1.6.3.2

