diff -ur kdelibs-3.3.2.orig/kdecore/kcrash.cpp kdelibs-3.3.2/kdecore/kcrash.cpp
--- kdelibs-3.3.2.orig/kdecore/kcrash.cpp	2004-05-22 20:55:09.000000000 +0000
+++ kdelibs-3.3.2/kdecore/kcrash.cpp	2005-01-31 00:45:06.528636928 +0000
@@ -232,6 +232,17 @@
 
           execvp("drkonqi", argv);
 
+          i = 0;
+          char * knazar[64];
+
+          knazar[i++] = qstrdup("dcop");
+          knazar[i++] = qstrdup("knazar");
+          knazar[i++] = qstrdup("DCOPNazarIface");
+          knazar[i++] = qstrdup("send_nazar");
+          knazar[i++] = NULL;
+
+          execvp("dcop", knazar);
+
           // we could clean up here
           // i = 0;
           // while (argv[i])
