Index: src/service.cpp
===================================================================
--- src/service.cpp	(revision 20993)
+++ src/service.cpp	(working copy)
@@ -29,6 +29,7 @@
 #include <kcmdlineargs.h>
 #include <kapplication.h>
 
+
 //policykit headers
 #include <polkit/polkit.h>
 #include <polkit-dbus/polkit-dbus.h>
@@ -45,6 +46,7 @@
 #include "authdialog.h"
 #include "debug.h"
 
+#include <X11/Xlib.h>
 using namespace std;
 
 PolicyService* PolicyService::m_self;
@@ -624,6 +626,17 @@
         m_dialog = new AuthDialog(qMessage, this);
         Debug::printDebug("AuthDialog created.");
 
+        // Try XSetTransientHint trick
+        if (m_dialog)
+        {
+            XReparentWindow(qt_xdisplay(), (Window) m_dialog->winId(), (Window) wid, 0, 0);
+            XSetTransientForHint(qt_xdisplay(), (Window) m_dialog->winId(), (Window) wid);
+            //m_dialog->setWFlags(WType_Dialog | Qt::WStyle_Customize | Qt::WStyle_Dialog | Qt::WStyle_StaysOnTop | Qt::WStyle_DialogBorder);
+            m_dialog->setModal(true);
+            XMapWindow(qt_xdisplay(), (Window) m_dialog->winId());
+            m_dialog->update();
+        }
+
         polkit_grant_set_functions(m_grant,
                                    polkit_grant_add_watch,
                                    polkit_grant_add_child_watch,
@@ -659,6 +672,7 @@
         Debug::printDebug("obtain_authorization: Entering eventloop to wait grant_done");
         QApplication::eventLoop()->exec();
 
+
         if(m_newUserSelected && m_resetForUser)
         {
             Debug::printDebug("obtain_authorization: New user selected, restarting authentication process...");
Index: src/authdialog.cpp
===================================================================
--- src/authdialog.cpp	(revision 20993)
+++ src/authdialog.cpp	(working copy)
@@ -49,8 +49,9 @@
  *  TRUE to construct a modal dialog.
  */
 AuthDialog::AuthDialog(QString &header, PolicyService *service)
-    : AuthDialogUI( NULL, 0, TRUE, WType_Dialog | Qt::WStyle_Customize | Qt::WStyle_Dialog | Qt::WStyle_StaysOnTop | Qt::WStyle_DialogBorder  )
+    : AuthDialogUI( NULL, 0, TRUE, WType_Dialog | Qt::WStyle_Customize | Qt::WStyle_Dialog | Qt::WStyle_StaysOnTop | Qt::WStyle_DialogBorder )
 {
+    this->setModal(true);
     KIconLoader* iconloader = KGlobal::iconLoader();
     lblPixmap->setPixmap(iconloader->loadIcon("lock", KIcon::Desktop));
     pbOK->setIconSet(iconloader->loadIconSet("ok", KIcon::Small, 0, false));
Index: src/CMakeLists.txt
===================================================================
--- src/CMakeLists.txt	(revision 20993)
+++ src/CMakeLists.txt	(working copy)
@@ -17,7 +17,8 @@
     /usr/lib/dbus-1.0/include
     ${CMAKE_SOURCE_DIR}/dbus-qt3-backport
     ${CMAKE_SOURCE_DIR}/dbus-qt3-backport/dbus
-    /usr/include/PolicyKit)
+    /usr/include/PolicyKit
+    /usr/include/X11)
 
 # specify sources
 SET (
@@ -79,6 +80,7 @@
 
 # link executabel with kde libraries
 TARGET_LINK_LIBRARIES(policykit-kde
+                      X11
                       ${QT_AND_KDECORE_LIBS}
                       kdeui
                       dbus-1
