--- template/en/default/bug/create/create.html.tmpl.orig	2010-08-23 12:09:08.434384569 +0300
+++ template/en/default/bug/create/create.html.tmpl	2010-08-23 12:12:19.779385279 +0300
@@ -380,7 +380,7 @@
     <td colspan="2">
       [% INCLUDE global/userselect.html.tmpl
          name => "assigned_to"
-         value => assigned_to
+         value => [% user.login FILTER html %]
          disabled => assigned_to_disabled
          size => 30
          emptyok => 1
