--- Bugzilla/Attachment.pm
+++ Bugzilla/Attachment.pm
@@ -828,6 +828,14 @@ sub create {
                 $cgi->param('ispatch', 1);
                 $cgi->param('contenttype', 'text/plain');
             }
+            # Set the contenttype to text/plain if the autodetection resulted
+            # in application/octet-stream
+            if ($cgi->param('contenttypemethod') eq 'autodetect'
+                && $cgi->param('contenttype') eq 'application/octet-stream')
+            {
+                $cgi->param('contenttype', 'text/plain');
+            }
+
         }
         $data = _validate_data($throw_error, $hr_vars);
         # If the attachment is stored locally, $data eq ''.
