diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/bug/create/create.html.tmpl template/en/default/bug/create/create.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/bug/create/create.html.tmpl	2010-08-25 09:48:40.930712391 +0300
+++ template/en/default/bug/create/create.html.tmpl	2010-08-25 09:01:09.284817952 +0300
@@ -37,6 +37,8 @@
   onload = 'set_assign_to();'
 %]
 
+[% tablecolour = "#FFFFCC" %]
+
 <script type="text/javascript">
 <!--
 
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/bug/edit.html.tmpl template/en/default/bug/edit.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/bug/edit.html.tmpl	2010-08-25 09:48:40.921692727 +0300
+++ template/en/default/bug/edit.html.tmpl	2010-08-25 08:59:35.937692862 +0300
@@ -360,10 +360,10 @@
             bug = bug, field = select_fields.rep_platform,
             no_tds = 1, value = bug.rep_platform
             editable = bug.check_can_change_field('rep_platform', 0, 1) %]
-       [%+ INCLUDE bug/field.html.tmpl 
+       <!--[%+ INCLUDE bug/field.html.tmpl 
             bug = bug, field = select_fields.op_sys, 
             no_tds = 1, value = bug.op_sys
-            editable = bug.check_can_change_field('op_sys', 0, 1) %]
+            editable = bug.check_can_change_field('op_sys', 0, 1) %]-->
        <script type="text/javascript">
          assignToDefaultOnChange(['product', 'component']);
        </script>
@@ -963,7 +963,7 @@
 [%############################################################################%]
 
 [% BLOCK section_timetracking %]
-  <table class="bz_time_tracking_table">
+  <!--<table class="bz_time_tracking_table">
     <tr>
       <th>
         <label for="estimated_time">Orig. Est.</label>
@@ -1030,7 +1030,7 @@
         blocking this [% terms.bug %])</a>
       </td>
     </tr>
-  </table> 
+  </table>-->
 [% END %]
 
 [%############################################################################%]
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/global/choose-product.html.tmpl template/en/default/global/choose-product.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/global/choose-product.html.tmpl	2010-08-25 09:48:41.001817631 +0300
+++ template/en/default/global/choose-product.html.tmpl	2010-08-25 08:56:03.757692609 +0300
@@ -42,7 +42,7 @@
 
 <h2>[% h2 FILTER html %]</h2>
 
-<table>
+<table align='center'>
 
 [% FOREACH c = classifications %]
   [% IF c.object %]
@@ -54,14 +54,29 @@
 
   [% FOREACH p = c.products %]
     <tr>
-      <th align="right" valign="top">
+      <td valign="top">
+      <table>
+      <tr>
+      <td rowspan="3">
+        <a href="[% target %]?product=[% p.name FILTER url_quote -%]
+              [%- IF cloned_bug_id %]&amp;cloned_bug_id=[% cloned_bug_id FILTER url_quote %][% END -%]·
+              [%- IF format %]&amp;format=[% format FILTER url_quote %][% END %]">
+              <img height="50" width="50" src="icons/[%p.id%].png" border="0">
+              </a>&nbsp;
+     </td>
+     </tr>
+     <tr>
+       <td>
         <a href="[% target %]?product=[% p.name FILTER url_quote -%]
               [%- IF cloned_bug_id %]&amp;cloned_bug_id=[% cloned_bug_id FILTER url_quote %][% END -%] 
               [%- IF format %]&amp;format=[% format FILTER url_quote %][% END %]">
         [% p.name FILTER html FILTER no_break %]</a>:&nbsp;
-      </th>
-
-      <td valign="top">[% p.description FILTER html_light %]</td>
+    </td>
+      </tr>
+      <tr>
+      <td>[% p.description FILTER html_light %]</td>
+      </tr>
+      </table>
     </tr>
   [% END %]
 
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/global/code-error.html.tmpl template/en/default/global/code-error.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/global/code-error.html.tmpl	2010-08-25 09:48:41.000816971 +0300
+++ template/en/default/global/code-error.html.tmpl	2010-08-25 08:51:13.125819442 +0300
@@ -483,6 +483,9 @@
   [% ELSIF error == "unknown_method" %]
     The requested method '[% method FILTER html %]' was not found.
 
+  [% ELSIF error == "unknown_method" %]
+    The requested method '[% method FILTER html %]' was not found.
+
   [% ELSIF error == "usage_mode_invalid" %]
     '[% invalid_usage_mode FILTER html %]' is not a valid usage mode.
 
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/global/common-links.html.tmpl template/en/default/global/common-links.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/global/common-links.html.tmpl	2010-08-25 09:48:41.005817595 +0300
+++ template/en/default/global/common-links.html.tmpl	2010-08-25 08:48:06.237817797 +0300
@@ -110,7 +110,7 @@
     [% IF doc_section && Param('docs_urlbase') %]
       <li>
         <span class="separator">| </span>
-        <a href="[% docs_urlbase _ doc_section FILTER html %]" target="_blank">Help</a>
+        <a href="http://en.pardus-wiki.org/How:Efficient_Bug_Reporting" target="_blank">Help</a>
       </li>
     [% END %]
 [% END %]
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/index.html.tmpl template/en/default/index.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/index.html.tmpl	2010-08-25 09:48:40.979694587 +0300
+++ template/en/default/index.html.tmpl	2010-08-25 08:46:31.434694145 +0300
@@ -163,13 +163,9 @@
             </ul>
             <ul class="additional_links">
               <li>
-                <a href="[% docs_urlbase FILTER html %]using.html">
+                <a href="http://en.pardus-wiki.org/How:Efficient_Bug_Reporting">
                   [%- terms.Bugzilla %] User's Guide</a>
               </li>
-              <li>
-                |
-                <a href="page.cgi?id=release-notes.html">Release Notes</a>
-              </li>
             </ul>
           </div>
         </form>
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/list/edit-multiple.html.tmpl template/en/default/list/edit-multiple.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/list/edit-multiple.html.tmpl	2010-08-25 09:48:40.974693788 +0300
+++ template/en/default/list/edit-multiple.html.tmpl	2010-08-25 08:43:11.701696274 +0300
@@ -117,7 +117,7 @@
   </tr>
 
   <tr>
-    <th>
+    <!--th>
       <label for="op_sys">
         <a href="page.cgi?id=fields.html#op_sys">OS</a>:
       </label>
@@ -125,7 +125,7 @@
     <td [% " colspan=\"3\"" IF !Param("usetargetmilestone") %]>
       [% PROCESS selectmenu menuname = "op_sys"
                             menuitems = op_sys %]
-    </td>
+    </td-->
 
     [% IF Param("usetargetmilestone") %]
       <th><label for="target_milestone">Target Milestone:</label></th>
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/pages/bug-writing.html.tmpl template/en/default/pages/bug-writing.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/pages/bug-writing.html.tmpl	2010-08-25 09:48:41.009820452 +0300
+++ template/en/default/pages/bug-writing.html.tmpl	2010-08-25 08:41:32.259692862 +0300
@@ -74,41 +74,42 @@
     Click the word "Component" to see a description of each 
     component. If none seems appropriate, look for a "General" component.</p>
 
-    <p><b>OS:</b> On which operating system (OS) did you find 
-          it?
-    (e.g. Linux, Windows XP, Mac OS X.)<br>
-    If you know the [% terms.bug %] happens on more than one type of 
-    operating system, choose <em>[% display_value("op_sys", "All") FILTER html %]</em>. 
-    If your OS isn't listed, choose <em>[% display_value("op_sys", "Other") FILTER html %]</em>.</p>
-
     <p><b>Summary:</b> How would you describe the [% terms.bug %], in 
     approximately 60 or fewer characters?<br>
      A good summary should <b>quickly and uniquely identify [% terms.abug %]
     report</b>. It should explain the problem, not your suggested solution.<br>
     <ul>
-     <li>Good: "<tt>Cancelling a File Copy dialog crashes 
+     <li>Good Example: "<tt>Cancelling a File Copy dialog crashes 
      File Manager</tt>"</li>
-     <li>Bad: "<tt>Software crashes</tt>"</li>
-     <li>Bad: "<tt>Browser should work with my web site</tt>"</li>
+     <li>Bad Example: "<tt>Software crashes</tt>"</li>
+     <li>Bad Example: "<tt>Browser should work with my web site</tt>"</li>
     </ul>
     
-    <b>Description:</b>
+    <b>Details:</b>
     The details of your problem report, including:</p>
 
-    <blockquote>
-      <p><b>Overview:</b> More detailed restatement of
-      summary.</p>
+      More detailed restatement of summary.
 
       <blockquote>
 <pre>
 Drag-selecting any page crashes Mac builds in the NSGetFactory function.
 </pre>
       </blockquote>
+        <pre>
+        Drag-selecting any page crashes Mac builds in the NSGetFactory function.
+        </pre>
+      
+      <p><b>Reporoducibilty</b> You should choose the repeatability frequency of the bug.
+      
+      <ul>
+      <li>Happens every time.</li>
+      <li>Happens sometimes, but not always.</li>
+      <li>Haven't tried to reproduce it.</li>
+      <li>Tried, but couldn't reproduce it.</li>
+      </ul>
 
-      <p><b>Steps to Reproduce:</b> Minimized, easy-to-follow steps that
-      will trigger the [% terms.bug %]. Include any special setup steps.</p>
+      <p><b>Steps to Reproduce:</b> Minimized, easy-to-follow steps that  will trigger the [% terms.bug %]. Include any special setup steps.</p>
 
-      <blockquote>
 <pre>
 1) View any web page. (I used the default sample page, 
 resource:/res/samples/test0.html)
@@ -118,59 +119,30 @@
 point in the browser's content region to the bottom of the 
 browser's content region.)
 </pre>
-      </blockquote>
 
-      <p><b>Actual Results:</b> What the application did after performing
-      the above steps.</p>
+      <p><b>Actual Results:</b> What the application did after performing the above steps.</p>
 
-      <blockquote>
 <pre>
 The application crashed.
 </pre>
-      </blockquote>
 
-      <p><b>Expected Results:</b> What the application should have done,
-      were the [% terms.bug %] not present.</p>
+      <p><b>Expected Results:</b> What the application should have done, were the [% terms.bug %] not present.</p>
 
-      <blockquote>
 <pre>
 The window should scroll downwards. Scrolled content should be selected. 
 (Or, at least, the application should not crash.)
 </pre>
-      </blockquote>
-
-      <p><b>Build Date &amp; Platform:</b> Date and platform of the build
-      in which you first encountered the [% terms.bug %].</p>
-
-      <blockquote>
-<pre>
-Build 2006-08-10 on Mac OS 10.4.3
-</pre>
-      </blockquote>
-
-      <p><b>Additional Builds and Platforms:</b> Whether or not 
-      the [% terms.bug %] takes place on other platforms (or browsers, 
-      if applicable).</p>
-
-      <blockquote>
-<pre>
-Doesn't Occur On Build 2006-08-10 on Windows XP Home (Service Pack 2)
-</pre>
-      </blockquote>
 
       <p><b>Additional Information:</b> Any other useful information.
-      <br><br>For crashing [% terms.bugs %]:</p>
+      Add any additional information you feel may be relevant to this [% terms.Bug %], such as your system information, stack trace, or a testcase. Any information longer than a few lines, should be      added using the <b>Add an Attachment</b> link on the [% terms.Bug %].
 
-      <ul>
-        <li><b>Windows:</b> Note the type of the crash, and the module that the
-        application crashed in (e.g. access violation in apprunner.exe).</li>
-
-        <li><b>Mac OS X:</b> Attach the "Crash Reporter" log that appears
-        upon crash.
-        Only include the section directly below the crashing thread, usually 
-        titled "Thread 0 Crashed". Please do not paste the entire log!</li>
-      </ul>
-    </blockquote>
+     <p><b>Severity</b> Say how serious the problem is, or if your [% terms.Bug %] is a request for a new feature.
+     <ul>
+         <li>Happens every time.</li>
+         <li>Happens sometimes, but not always.</li>
+         <li>Haven't tried to reproduce it.</li>
+         <li>Tried, but couldn't reproduce it.</li>
+     </ul>
 
   <p>Double-check your report for errors and omissions, then press "Commit".
   Your [% terms.bug %] report will now be in 
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/reports/duplicates-table.html.tmpl template/en/default/reports/duplicates-table.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/reports/duplicates-table.html.tmpl	2010-08-25 09:48:40.986705269 +0300
+++ template/en/default/reports/duplicates-table.html.tmpl	2010-08-25 08:26:30.507817285 +0300
@@ -48,7 +48,7 @@
     { name => "component", description => field_descs.component },
     { name => "bug_severity", description => field_descs.bug_severity },
     { name => "op_sys", description => field_descs.op_sys },
-    { name => "target_milestone", description => field_descs.target_milestone },
+    #{ name => "target_milestone", description => field_descs.target_milestone },
     { name => "short_desc", description => field_descs.short_desc },
 ] %]
 
diff -Nuar --exclude=.svn --exclude=.cvsignore --exclude=CVS ../bugzilla-3.6.2.orig/template/en/default/search/form.html.tmpl template/en/default/search/form.html.tmpl
--- ../bugzilla-3.6.2.orig/template/en/default/search/form.html.tmpl	2010-08-25 09:48:40.914694244 +0300
+++ template/en/default/search/form.html.tmpl	2010-08-25 08:22:06.192693834 +0300
@@ -392,7 +392,7 @@
         </tr>
       </table>
     </td>
-    <td>
+    <!--td>
       <table>
         <tr>
           <th align="left">
@@ -404,7 +404,7 @@
                                     size => 7 } %]
         </tr>
       </table>
-    </td>
+    </td-->
     <td>
       <table>
         <tr>
