--- install-radius-db.PL.old	2010-08-16 16:14:53.483057313 +0300
+++ install-radius-db.PL	2010-08-16 16:15:03.207057064 +0300
@@ -1,7 +1,7 @@
 #!/usr/bin/perl -w
 # $Id: install-radius-db.PL,v 1.2 2004/12/18 04:51:17 andrew Exp $
 my $source_dir = 'raddb';
-my $raddb_dir = '/etc/raddb';
+my $raddb_dir = '/var/pisi/perl-RadiusPerl-0.17-1/install/etc/raddb';
 
 unless (-w $raddb_dir) {
 	# Cannot write to the raddb directory
