diff -Nuar PackageKit-0.4.9-orig/src/Makefile.am PackageKit-0.4.9/src/Makefile.am
--- PackageKit-0.4.9-orig/src/Makefile.am	2009-08-12 09:40:16.612986410 +0300
+++ PackageKit-0.4.9/src/Makefile.am	2009-08-13 13:38:07.289777394 +0300
@@ -220,7 +220,7 @@
 install-data-hook:
 	if test -w $(DESTDIR)$(prefix)/; then \
 		mkdir -p $(DESTDIR)$(localstatedir)/cache/PackageKit/downloads; \
-		mkdir -p $(PK_LOG_DIR); \
+		mkdir -p $(DESTDIR)$(PK_LOG_DIR); \
 	fi
 
 EXTRA_DIST =						\
diff -Nuar PackageKit-0.4.9-orig/src/Makefile.in PackageKit-0.4.9/src/Makefile.in
--- PackageKit-0.4.9-orig/src/Makefile.in	2009-08-12 09:40:16.613986071 +0300
+++ PackageKit-0.4.9/src/Makefile.in	2009-08-13 13:37:31.968777366 +0300
@@ -2131,7 +2131,7 @@
 install-data-hook:
 	if test -w $(DESTDIR)$(prefix)/; then \
 		mkdir -p $(DESTDIR)$(localstatedir)/cache/PackageKit/downloads; \
-		mkdir -p $(PK_LOG_DIR); \
+		mkdir -p $(DESTDIR)$(PK_LOG_DIR); \
 	fi
 
 clean-local:
