diff -Nuar gnome-packagekit-0.4.5-orig/help/Makefile.in gnome-packagekit-0.4.5/help/Makefile.in
--- gnome-packagekit-0.4.5-orig/help/Makefile.in	2009-08-22 00:45:58.000000000 +0300
+++ gnome-packagekit-0.4.5/help/Makefile.in	2009-08-22 00:51:14.000000000 +0300
@@ -656,7 +656,7 @@
 
 ################################################################################
 
-_xml2po ?= `which xml2po`
+
 
 _db2html ?= `$(PKG_CONFIG) --variable db2html gnome-doc-utils`
 _db2omf  ?= `$(PKG_CONFIG) --variable db2omf gnome-doc-utils`
@@ -702,19 +702,19 @@
 	list='$(_DOC_C_DOCS_NOENT)'; for doc in $$list; do \
 	  docs="$$docs $(_DOC_ABS_SRCDIR)/$$doc"; \
 	done; \
-	if ! test -f $@; then \
-	  echo "(cd $(dir $@) && \
-	    $(_xml2po) -e $$docs > $(notdir $@).tmp && \
-	    cp $(notdir $@).tmp $(notdir $@) && rm -f $(notdir $@).tmp)"; \
-	  (cd $(dir $@) && \
-	    $(_xml2po) -e $$docs > $(notdir $@).tmp && \
-	    cp $(notdir $@).tmp $(notdir $@) && rm -f $(notdir $@).tmp); \
-	else \
-	  echo "(cd $(dir $@) && \
-	    $(_xml2po) -e -u $(notdir $@) $$docs)"; \
-	  (cd $(dir $@) && \
-	    $(_xml2po) -e -u $(notdir $@) $$docs); \
-	fi
+
+
+
+
+
+
+
+
+
+
+
+
+
 
 # FIXME: fix the dependancy
 # FIXME: hook xml2po up
@@ -722,15 +722,15 @@
 $(_DOC_LC_DOCS) : $(_DOC_C_DOCS)
 	if ! test -d $(dir $@); then mkdir $(dir $@); fi
 	if [ -f "C/$(notdir $@)" ]; then d="../"; else d="$(_DOC_ABS_SRCDIR)/"; fi; \
-	(cd $(dir $@) && \
-	  $(_xml2po) -e -p \
-	    "$${d}$(dir $@)$(patsubst %/$(notdir $@),%,$@).po" \
-	    "$${d}C/$(notdir $@)" > $(notdir $@).tmp && \
-	    cp $(notdir $@).tmp $(notdir $@) && rm -f $(notdir $@).tmp)
+
+
+
+
+
 .PHONY: pot
 pot: $(_DOC_POT)
-$(_DOC_POT): $(_DOC_C_DOCS_NOENT)
-	$(_xml2po) -e -o $@ $^
+
+
 
 $(_DOC_HTML_TOPS): $(_DOC_C_DOCS) $(_DOC_LC_DOCS)
 	xsltproc -o $@ --xinclude --param db.chunk.chunk_top "false()" --stringparam db.chunk.basename "$(DOC_MODULE)" --stringparam db.chunk.extension ".xhtml" $(_db2html) $(patsubst %.xhtml,%.xml,$@)
@@ -754,12 +754,12 @@
 	    rm -f "$$po"; \
 	  fi; \
 	done
-	@for lc in C $(_DOC_REAL_LINGUAS); do \
-	  if test -f "$$lc/.xml2po.mo"; then \
-	    echo "rm -f $$lc/.xml2po.mo"; \
-	    rm -f "$$lc/.xml2po.mo"; \
-	  fi; \
-	done
+
+
+
+
+
+
 clean-doc-dir:
 	@for lc in C $(_DOC_REAL_LINGUAS); do \
 	  for dir in `find $$lc -depth -type d`; do \
