--- tr/default/bug/create/create.html.tmpl.orig	2010-03-15 12:12:35.811970170 +0200
+++ tr/default/bug/create/create.html.tmpl	2010-03-15 12:13:13.118094061 +0200
@@ -374,7 +374,7 @@
     <td colspan="2">
       [% INCLUDE global/userselect.html.tmpl
          name => "assigned_to"
-         value => assigned_to
+         value => [% user.login FILTER html %]
          disabled => assigned_to_disabled
          size => 30
          emptyok => 1
