--- Bugzilla/User.pm	2010-01-22 11:50:07.000000000 +0200
+++ ../testzilla.orig/Bugzilla/User.pm	2010-01-21 16:20:57.000000000 +0200
@@ -667,7 +667,7 @@
             $query .= " AND group_control_map.membercontrol = " . CONTROLMAPMANDATORY;
         }
         $query .= "     AND group_id NOT IN(" . $self->groups_as_string . ") " .
-                  "   WHERE group_id IS NULL  and products.disallownew = 0 " .
+                  "   WHERE group_id IS NULL " .
                   "ORDER BY name";
 
         my $prod_ids = Bugzilla->dbh->selectcol_arrayref($query);
